src/NumericalAlgorithms/Interpolation/CubicSpline.hpp

```diff
@@ -0,0 +1,60 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <gsl/gsl_spline.h>
+#include <vector>
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace intrp {
+/*!
+ * \ingroup NumericalAlgorithmsGroup
+ * \brief A cubic spline interpolation class
+ *
+ * The class builds a cubic spline interpolant using the `x_values` and
```

Comment:
what type of spline? Is it a natural cubic or is there a way to specify derivative values at end points? If GSL offers the latter than that functionality should be added.

Reply:
it's natural cubic